core/src/main/java/org/infinispan/marshall/persistence/impl/PersistenceMarshallerImpl.java

```diff
@@ -93,6 +98,19 @@ private Marshaller createUserMarshaller() {
       }
    }
 
+   private Marshaller createProtoStreamUserMarshaller(SerializationConfiguration serializationConfig) {
+      // TODO is it overkill to have separate SerializationContext for persistence AND user?
```

Comment:
Interesting question. I'd say we can have a single context for both user marshalling and persistence marshalling. If any separation is needed that is achieved via packages.

Reply:
I agree, otherwise we're introducing wrapping for very little benefit